### PR TITLE
[4.0] Searchtools mobile display

### PR DESCRIPTION
**Pull request by infograf768**
Follow-up on https://github.com/joomla/joomla-cms/pull/27416

### Summary of Changes
Take off from _toolbar.scss `.js-stools-container-selector` which interfere with searchtools.css
Correct margins in mobile view as the situation is different when we have `.js-stools-container-selector` (1 field) or `.js-stools-container-selector-first` (2 fields).

### Testing Instructions
Install multiple languages including Persian (fa-IR)
Display in Mobile view the Installed language manager and also the Menu Items manager.
These contains fields separated from the main searchtools (site/admin for Installed languages and site/admin // menu for menu items.

Patch and use npm
### Before patch
LTR

Screen Shot 2020-01-09 at 09 51 41

RTL

Screen Shot 2020-01-09 at 09 55 06

### After patch
LTR

Screen Shot 2020-01-09 at 09 12 14

RTL

Screen Shot 2020-01-09 at 09 11 26

@Fedik @brianteeman

### CODE REVIEW

**Comment by brianteeman** [comment]
quickcheck - looks great
will test properly later today

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 1]
```diff
@@ -106,6 +106,54 @@
...
+      width: 100%;
+
+      [dir=ltr] & {
+        margin-right: 0;
+      }
+
+      [dir=rtl] & {
+        margin-right: 0;
+      }
+    }
```
Comment (brianteeman) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss:114]: as the value is the same for LTR and RTL you should not need to have individual settings

Comment (infograf768) [reply administrator/templates/atum/scss/system/searchtools/searchtools.scss:114]: Agree. There are other places to also simplify. let's wait for more small changes to modify.

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 2]
```diff
@@ -106,6 +106,54 @@
...
+
+    @include media-breakpoint-down(xs) {
+      float: none !important;
```
Comment (ciar4n) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss]: Already defined under `media-breakpoint-down(sm)`

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 3]
```diff
@@ -106,6 +106,54 @@
...
+      width: 100%;
```
Comment (ciar4n) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss]: Already defined under `media-breakpoint-down(sm)`

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 4]
```diff
@@ -106,6 +106,54 @@
...
+
+      [dir=ltr] & {
+        margin-right: 0;
```
Comment (ciar4n) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss:118]: Already defined under `media-breakpoint-down(sm)`

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 5]
```diff
@@ -106,6 +106,54 @@
...
+      }
+
+      [dir=rtl] & {
+        margin-right: 0;
```
Comment (ciar4n) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss]: Already defined under `media-breakpoint-down(sm)`

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 6]
```diff
@@ -106,6 +106,54 @@
...
+        max-width: 100%;
+        margin-right: 0;
+
+        [dir=rtl] & {
+          margin-right: 0;
+        }
+      }
+
+      @include media-breakpoint-down(xs) {
+        max-width: 100%;
```
Comment (ciar4n) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss]: Already defined under `media-breakpoint-down(sm)`

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 7]
```diff
@@ -106,6 +106,54 @@
...
+        margin-right: .5rem;
+
+        [dir=rtl] & {
+          margin-right: 0;
```
Comment (ciar4n) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss]: Already defined under `media-breakpoint-down(sm)`

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 8]
```diff
@@ -143,6 +201,15 @@
           margin-right: 0;
         }
       }
+
+      @include media-breakpoint-down(xs) {
+        max-width: 100%;
```
Comment (ciar4n) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss]: Already defined under `media-breakpoint-down(sm)`

File: administrator/templates/atum/scss/system/searchtools/searchtools.scss [hunk 9]
```diff
@@ -143,6 +201,15 @@
...
+        margin-right: .5rem;
+
+        [dir=rtl] & {
+          margin-right: 0;
```
Comment (ciar4n) [inline administrator/templates/atum/scss/system/searchtools/searchtools.scss:185]: Already defined under `media-breakpoint-down(sm)`

**Comment by brianteeman** [comment]
It works but it needs cleaning up to prevent repetition

**Comment by infograf768** [comment]
@ciar4n

Alas I had already merged https://github.com/joomla/joomla-cms/pull/27443/commits/8b253059f967ca6d88f5434adade533d7c08f321

Therefore your proposals were hard to find out. 
Anyway I modified with most of your suggestions.
Except some that kill the alignment in LTR when we have 2 fields (menu items manager) for the second field. I had there to keep repeating `margin-right: 0;`

Please look again and test. If you propose some more changes, please test them before, following my instructions.

**Comment by ciar4n** [comment]
@infograf768 

It looks like you got them all.

On a side note, there is a couple of ways of improving this...

1. **mobile first** - only apply the margins to desktops `media-breakpoint-up(sm)` rather than applying margins to all and then taking it away on small screens.

2. Create a unique class for all toolbar fields and only define changes once.

**Comment by infograf768** [comment]
@ciar4n 
Not sure at all what you mean as xs may need different margins than sm

We have 3 solutions:
1. Keep this, test it OK and merge. Then make a new PR.
2. Kill this PR and someone makes a new PR
3. Propose a PR to my branch.

**Comment by ciar4n** [comment]
> Not sure at all what you mean as xs may need different margins than sm

On a single column layout there is no reason why they should.  The frontend template CSS is mobile first, which is why I suggested it here.

**Comment by infograf768** [comment]
Then please make patch.

**Comment by ciar4n** [comment]
I have tested this item :white_check_mark: successfully on 118dc6f5c112e7bd76867910dea4569325e1081b<br /><br />This works. Suggestions can be applied template wide in a separate pr (if required).<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27443">issues.joomla.org/tracker/joomla-cms/27443</a>.</sub>

**Comment by infograf768** [comment]
@ciar4n 
thanks for testing
_BTW, I can demonstrate easily why we will have different margin-right when we are in LTR and 2 fields are concerned. xs needs `margin-right: .5rem` while sm needs `margin-right: 0;`
Can send you an animated gif if you desire._

@brianteeman
Please mark your test so that we merge this.

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 118dc6f5c112e7bd76867910dea4569325e1081b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27443">issues.joomla.org/tracker/joomla-cms/27443</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27443">issues.joomla.org/tracker/joomla-cms/27443</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**